src/memory/memory.c

```diff
@@ -710,3 +727,77 @@ bool memory_add_noise_remote_static_pubkey(const uint8_t* pubkey)
 
     return _write_chunk(CHUNK_1, chunk.bytes);
 }
+
+memory_multisig_set_by_hash_result_t memory_multisig_set_by_hash(
+    const uint8_t* hash,
+    const char* name)
+{
+    chunk_2_t chunk = {0};
+    CLEANUP_CHUNK(chunk);
+    _read_chunk(CHUNK_2, chunk_bytes);
+
+    bool found = false;
+    size_t write_index = 0;
+    size_t empty_index = 0;
+    bool empty_found = false;
+
+    uint8_t empty[32];
+    memset(empty, 0xFF, sizeof(empty));
+
+    // Error if there is already a different multisig config with the same name.
+    for (size_t i = 0; i < MEMORY_MULTISIG_NUM_ENTRIES; i++) {
+        const multisig_configuration_t* multisig = &chunk.fields.multisig_configs[i];
+        if (STREQ(multisig->name, name)) {
+            if (!MEMEQ(multisig->hash, hash, sizeof(multisig->hash))) {
+                return MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_DUPLICATE_NAME;
+            }
+            // config already exists (equal), early abort, skipping another write.
+            return MEMORY_MULTISIG_SET_BY_HASH_RESULT_OK;
+        }
+    }
+
```

Comment:
Could you write a comment that this loop is doing two things. Both find the *first* empty index `empty_index` and find the index of the hash if the hash is present and select that for storing into.

Reply:
Done